index.js

```diff
@@ -396,28 +423,43 @@ SendStream.prototype.isRangeFresh = function isRangeFresh(){
  * @private
  */
 
-SendStream.prototype.redirect = function redirect(path) {
+SendStream.prototype.redirect = function redirectDirectory(path) {
   if (listenerCount(this, 'directory') !== 0) {
     this.emit('directory')
     return
   }
 
   if (this.hasTrailingSlash()) {
-    this.error(403)
-    return
+    return this.error(403)
```

Comment:
I should have missed it before. Anyway it doesn't matter if it returns the event-emitter or undefined as far as it's not being used, and in fact I've seen in several places that this is the recomended way to do it. Anyway, I've already fixed it.